### PR TITLE
Feat/civ 5680 | CCJ repayment plan

### PR DESCRIPTION
**Pull request by lfazelyniaHMCTS**
### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-5680

### Change description ###
Adding a screen to show payment options for repayment, containing the payment amount, first payment date and frequency of payment

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by laarmada — CHANGES_REQUESTED** [review]

File: src/main/modules/i18n/locales/cy.json [hunk 1]
```diff
@@ -316,6 +316,15 @@
...
     },
+    "CCJ_REPAYMENT_PLAN_INSTALMENTS": {
+      "EACH_INSTALMENT_AMOUNT": "Amount for each instalment",
+      "FIRST_PAYMENT_DATE": "Date for first payment",
+      "HOW_OFTEN": "How often do you want them to pay instalments?",
+      "INSTALMENT_AMOUNT_HELP": "For example £10",
+      "PAGE_TITLE": "Suggest instalments for the defendant",
+      "TITLE": "Suggest instalments for the defendant",
+      "TOTAL_CLAIM_AMOUNT": "Total claim amount is £{{ totalClaimAmount }}"
+    },
```
[the remark this answers is not in the document]
Comment (lfazelyniaHMCTS) [reply src/main/modules/i18n/locales/cy.json:327]: I've just checked, they are

File: src/main/routes/features/claimantResponse/ccj/repaymentPlanInstalmentsController.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      req.body.amount,
+      new InstalmentFirstPaymentDate(req.body.firstPaymentDate),
+      req.body.paymentFrequency,
+    ));
+    form.validateSync();
+
+    if (form.hasErrors()) {
+      renderView(form, claim.totalClaimAmount, res);
+    } else {
+      await saveClaimantResponse(claimId, form.model, 'repaymentPlanInstalments', 'ccjRequest');
```
[the remark this answers is not in the document]
Comment (lfazelyniaHMCTS) [reply src/main/routes/features/claimantResponse/ccj/repaymentPlanInstalmentsController.ts:53]: Just for my understanding, why should we use a const here when these values are only used once?
Seems a bit redundant no and wasted lines of code?

File: src/main/common/models/claimantResponse/ccj/paymentFrequencyType.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+export enum PaymentFrequencyType {
+  EACH_WEEK = 'EACH_WEEK',
```
[the remark this answers is not in the document]
Comment (lfazelyniaHMCTS) [reply src/main/common/models/claimantResponse/ccj/paymentFrequencyType.ts:2]: transactionSchedule is slightly different and isn't called EACH_WEEK (for example), but WEEK
Will thist affect things?

File: src/main/common/models/claimantResponse/ccj/repaymentPlanInstalments.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import {PaymentFrequencyType} from './paymentFrequencyType';
+import {toNumberOrUndefined} from '../../../../common/utils/numberConverter';
+
+export class RepaymentPlanInstalments {
+  @IsNumber({allowNaN: false, maxDecimalPlaces: 2}, {message: 'ERRORS.VALID_TWO_DECIMAL_NUMBER'})
+  @Min(0, {message: 'ERRORS.VALID_AMOUNT'})
+    amount?: number;
+
+  @ValidateNested()
+    firstPaymentDate?: InstalmentFirstPaymentDate;
```
[the remark this answers is not in the document]
Comment (lfazelyniaHMCTS) [reply src/main/common/models/claimantResponse/ccj/repaymentPlanInstalments.ts:12]: I can do actually, but the constructor on InstalmentFirstPaymentDate makes it cleaner on the controller when compared to PaymentDate
Better to reuse existing or create something (this model) that is cleaner?

File: src/main/routes/features/claimantResponse/ccj/repaymentPlanInstalmentsController.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (lfazelyniaHMCTS) [reply src/main/routes/features/claimantResponse/ccj/repaymentPlanInstalmentsController.ts:53]: And in this PR, it has been approved with string values, not consts. So shouldn't this be consistent with other CCJ controllers?
https://github.com/hmcts/civil-citizen-ui/pull/1119/files#diff-3078f8a9770df91d8aa3fa5d9c5e2242d501913ec41bd1c29a5214edb249a9d8

File: src/main/common/models/claimantResponse/ccj/repaymentPlanInstalments.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (lfazelyniaHMCTS) [reply src/main/common/models/claimantResponse/ccj/repaymentPlanInstalments.ts]: Resolved

**Review by portizdev — APPROVED** [review]

**Review by neeta-hmcts — APPROVED** [review]

File: src/main/common/models/claimantResponse/ccj/paymentFrequencyType.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (laarmada) [reply src/main/common/models/claimantResponse/ccj/paymentFrequencyType.ts:2]: but you dont use that enum for translation right? we can see on a call if you want...

File: src/main/routes/features/claimantResponse/ccj/repaymentPlanInstalmentsController.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (laarmada) [reply src/main/routes/features/claimantResponse/ccj/repaymentPlanInstalmentsController.ts:57]: yes!! I didnt review that one, the correct way is use constant :) the reviewers didnt view that detail....

File: src/main/common/models/claimantResponse/ccj/paymentFrequencyType.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (lfazelyniaHMCTS) [reply src/main/common/models/claimantResponse/ccj/paymentFrequencyType.ts]: Resolved

File: src/main/routes/features/claimantResponse/ccj/repaymentPlanInstalmentsController.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (lfazelyniaHMCTS) [reply src/main/routes/features/claimantResponse/ccj/repaymentPlanInstalmentsController.ts]: Resolved

**Review by laarmada — APPROVED** [review]

**Comment by lfazelyniaHMCTS** [comment]
Closing as pipeline build errors
Opening a new one as advised by Sampan

**Pull request closed without merging**